cinema/views.py

```diff
@@ -1 +1,81 @@
-# write views here
+from django.http import Http404
+from django.shortcuts import get_object_or_404
+from rest_framework.response import Response
+from rest_framework import status, generics, mixins, viewsets
+from .serializers import (MovieSerializer,
+                          ActorSerializer,
+                          MovieSessionSerializer,
+                          CinemaHallSerializer,
+                          GenreSerializer,
+                          MovieDetailSerializer,
+                          MovieListSerializer,
+                          MovieSessionListSerializer,
+                          MovieSessionDetailSerializer)
+from .models import (Genre,
+                     Actor,
+                     CinemaHall,
+                     Movie,
+                     MovieSession,
+                     Ticket,
+                     Order)
+
+
+class MovieViewSet(viewsets.ModelViewSet):
+    queryset = Movie.objects.all()
+    serializer_class = MovieDetailSerializer
+
+    def get_queryset(self):
+
+        queryset = self.queryset
+
+        if self.action == "list":
+            queryset = queryset.prefetch_related("actors", "genres")
+
+        return queryset
+
+    def get_serializer_class(self):
+        if self.action == "list":
+            return MovieListSerializer
+
+        if self.action == "retrieve":
+            return MovieDetailSerializer
+
+        return MovieSerializer
```

Comment:
`return self.serializer_class` is better

Reply:
in this case, serializer_class is a MovieDetailSerializer, so returning this serializer will cause an error